thorlcr/thorutil/thormisc.cpp

```diff
@@ -1480,3 +1480,37 @@ bool isRemoteReadCandidate(const CActivityBase &activity, const RemoteFilename &
     }
     return false;
 }
+
+void checkAndDumpAbortInfo(const char *cmd)
+{
+    try
+    {
+        StringBuffer dumpInfoCmd(cmd);
+        if (dumpInfoCmd.length())
+        {
+            /* add some params that might be useful to script
+             * 1) Thor instance name
+             * 2) base port
+             * 3) exe path
+             * 4) PID
+             */
+            const char *myInstanceName = globals->queryProp("@name");
+            unsigned myBasePort = getMachinePortBase();
+            StringBuffer exePath(queryCurrentProcessPath());
+            if (0 == exePath.length())
+                exePath.append("process-name-unknown");
+            unsigned pid = GetCurrentProcessId();
+            dumpInfoCmd.appendf(" %s %u %s %u", myInstanceName, myBasePort, exePath.str(), pid);
+        }
+        else
+            getDebuggerGetStacksCmd(dumpInfoCmd);
```

Comment:
So if there is a dumpInfoCmd configured then it is used instead of the gdb get all stacks ?
Does it always make sense to append thor-instance-name, base-port, exe-path, pid to dumpInfoCmd if we don't know what that is ?

Reply:
>So if there is a dumpInfoCmd configured then it is used instead of the gdb get all stacks ?

yes.

>Does it always make sense to append thor-instance-name, base-port, exe-path, pid to dumpInfoCmd if we don't know what that is ?

it should be documented that the command is always executed like this, but I think it's okay, unless you have another suggestion? 
An alternative approach I suppose would be to set interesting info. into environment variables for the executed program to use if it wanted to.